contrib/OtlpHttp/Exporter.php

```diff
@@ -0,0 +1,210 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Contrib\OtlpHttp;
+
+use InvalidArgumentException;
+use Opentelemetry\Proto\Collector\Trace\V1\ExportTraceServiceRequest;
+use OpenTelemetry\Sdk\Trace;
+use OpenTelemetry\Trace as API;
+use Psr\Http\Client\ClientExceptionInterface;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Client\NetworkExceptionInterface;
+use Psr\Http\Client\RequestExceptionInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\StreamFactoryInterface;
+
+class Exporter implements Trace\Exporter
+{
+    /**
+     * @var string
+     */
+    private $endpointUrl;
+
+    /**
+     * @var string
+     */
+    private $protocol;
+
+    /**
+     * @var string
+     */
+    private $insecure;
+
+    /**
+     * @var string
+     */
+    private $certificateFile;
+
+    /**
+     * @var string
+     */
+    private $headers;
+
+    /**
+     * @var string
+     */
+    private $compression;
+
+    /**
+     * @var int
+     */
+    private $timeout;
+    /**
+     * @var SpanConverter
+     */
+    private $spanConverter;
+
+    /**
+    * @var bool
+    */
+    private $running = true;
+
+    /**
+     * @var ClientInterface
+     */
+    private $client;
+
+    /**
+     * @var RequestFactoryInterface
+     */
+    private $requestFactory;
+
+    /**
+     * @var StreamFactoryInterface
+     */
+    private $streamFactory;
+
+    /**
+     * Exporter constructor.
+     */
+    public function __construct(
```

Comment:
Curious how we could pass config directly here rather than only via env vars. I did have a number of params here in the Grpc exporter but wasn't that great.

Maybe allow an optional array to be passed with: `['endpoint' => 'http://localhost/v1/traces', 'headers' => 'foo=bar']`. Would that be a clean interface?

Reply:
I think that'd be a clean interface as long as we document it well.  Thinking about how end users would implement this is an important part of the story.

Reply:
Other languages have named arguments which helps in this case. For us I think sufficient documentation and docstrings should go far. This can be added in a new feature.